pkg/proctree/datasource.go

```diff
@@ -118,7 +118,7 @@ func (ptds *DataSource) exportProcessInfo(
 		}
 		threadInfo := thread.GetInfo()
 		if threadInfo.IsAliveAt(queryTime) {
-			aliveThreads[threadInfo.GetPid()] = threadHash
+			aliveThreads[threadInfo.GetTid()] = threadHash
```

Comment:
I don't understand. Threads are of a process, and nit of another thread, no? So should we get the threads of some pid?

Reply:
This is a temporary map for the process data source query. Each query (for a single process) also provides a map of existing children (processes) and threads (lwps).

The fix is about indexing by thread PID. It was creating a single entry for multiple threads and replacing that entry over and over. That happened because all threads had the same PID but not the same TID.